### PR TITLE
Update documentation for versions 3.8+

### PR DESCRIPTION
**Pull request by niccokunzmann**
This compacts the documentation.
Mentions of old versions are removed.
A concise path from 5.x to 6.0 is outlined.
master -> main

Contributes to https://github.com/collective/icalendar/issues/513

### CODE REVIEW

File: README.rst
Comment (stevepiercy) [inline README.rst]: In combination with the replacing the compatibility field with a version badge, removing this line eliminates all text of Python version compatibility. That's not good. See suggestion below.

File: README.rst
Comment (stevepiercy) [inline README.rst]: We need to be explicit with version numbers of Python and use text. I used the trove classifiers in `setup.py` to get the specific versions. We might want to add 3.12, now that it has been released, but only if it passes tests.

```suggestion
Version 6 is on `branch main <https://github.com/collective/icalendar/>`_.
It is compatible with Python versions 3.7 - 3.11, and PyPy3.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: README.rst
Comment (niccokunzmann) [reply README.rst]: Since #645, we have a badge to list the versions. I would like to de-duplicate this.
The intention is to support all Python releases that are not end of life. How do you think we can improve this?

Comment (stevepiercy) [reply README.rst]: A badge only shows after rendered through the web. Even then it's not accessible for people with vision impairments and it lacks a context without an HTML `alt` attribute. It's not a satisfactory replacement, but it is a nice visual enhancement for people without vision impairments.

In a README.md, you should have a clear indicator of version support somewhere, whether that is explicit text, a statement to read the trove classifiers in `setup.py`, or a link to read documentation. IMO, it's most common and easiest to keep it in README.md, even though that means more maintenance and risk of duplication mistakes.

In documentation with Sphinx, you can use replacements to avoid maintenance and duplication mistakes. We do this in Plone. Example:

[conf.py](https://github.com/plone/documentation/blob/21453a6ea6c528bbe8bb0799d688d5dc2a5d1809/docs/conf.py#L348-L359)

In the documentation source files:

https://github.com/plone/documentation/blob/21453a6ea6c528bbe8bb0799d688d5dc2a5d1809/docs/install/create-project.md?plain=1#L76-L76

Which renders as https://6.docs.plone.org/install/create-project.html#pre-requisites-for-installation

If you wanted to go fully automatic, I suppose you could change the `setup(app)` to parse trove classifiers and set the result as the value for the replacement variable.

Comment (stevepiercy) [reply README.rst]: > The intention is to support all Python releases that are not end of life.

This should be documented at a policy level, and implemented throughout the repo wherever necessary.

I would go a step further, though, to future proof it. For example:

"icalendar supports all [released and supported versions of Python](https://devguide.python.org/versions/#supported-versions) at the time of its own release."

Thus if you released 6.0 today, then you would support Python 3.8-3.12 only.

In a future release, if you add or drop a version, then that could be a breaking change or a feature addition. You would follow your version numbering scheme accordingly when you make a release at that time.

File: README.rst
Comment (stevepiercy) [inline README.rst]: I just noticed that the issue name says 3.8+, but the trove classifiers and this line say 3.7-3.11. Which is correct?

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: README.rst
Comment (niccokunzmann) [reply README.rst]: For version 6, 3.8+ is correct. #634 is included in #630.

File: README.rst
Comment (niccokunzmann) [reply README.rst]: > It is compatible with Python versions 3.7 - 3.11, and PyPy3.

The above statement is in the README now.
If anything comes up, you can still add a PR.

File: README.rst
Comment (stevepiercy) [reply README.rst]: OK, then we will need to update #634 to include `CHANGES.rst` with the correct Python versions.

Comment (stevepiercy) [reply README.rst]: But wait a minute, look at the change log entry, the issue title, and this line of text. See the discrepancy in versions? They don't align with each other.

Comment (niccokunzmann) [reply README.rst]: I know. Please change it as you like it. I would like to be consistent between releases, not always. There are other PRs changing this at the same time.
It can be changed after merge, too. If you think, this is harmful, then, do not merge it. Otherwise I prefer progress over perfection.

Comment (stevepiercy) [reply README.rst]: I'll correct it in a review suggestion. I prefer to get details right as soon as I realize them, else I will likely forget to deal with them later.

Comment (stevepiercy) [reply README.rst]: Side note: I assume I cannot push to your remote, but I have not tried. For ease of collaboration where others can push to this remote (`collective/icalendar`), consider using this remote when you push.

File: CHANGES.rst
Comment (stevepiercy) [inline CHANGES.rst]:
```suggestion
- Change documentation to represent compatibility with Python 3.8 - 3.12, and PyPy3.
```

File: README.rst
Comment (stevepiercy) [inline README.rst]: Resolves Python version inconsistency with issue title and change log entry.

```suggestion
It is compatible with Python versions 3.8 - 3.12, and PyPy3.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Versions are now consistent with merge of suggestions, then OK to merge this PR.

**Review by stevepiercy — APPROVED** [review]
Ready to merge at your discretion @niccokunzmann.

File: README.rst
Comment (niccokunzmann) [reply README.rst]: Ok. This is a good suggestion to use this for collaboration. I sometimes create pull requests to other people's branches. I did not want to clutter what is going on here. Maybe, if I name that branches issue-..., then it is not so much of a clutter.

**Comment by niccokunzmann** [comment]
Nice, thanks for the documentation improvement!

**Pull request merged**
